src/chain/chain.rs

```diff
@@ -926,12 +913,19 @@ impl Chain {
         target: &Location,
         node_knowledge_override: Option<u64>,
     ) -> SectionProofSlice {
-        let first_index = match (target, node_knowledge_override) {
+        let first_index = self.knowledge_index(target, node_knowledge_override);
+        self.state.our_history.slice_from(first_index as usize)
+    }
+
+    /// Provide a start index of a SectionProofSlice that proves the given signature to the given
+    /// destination location.
+    /// If `node_knowledge_override` is `Some`, it is used when calculating proof for
```

Comment:
This second paragraph of the comment is bit duplicated with the one above function `prove` ?
maybe one of them can be removed?

Reply:
I don't really see why, essentially the effect of node_knowledge_override is the same for both public functions.
Unless we want something like:

```rust
\\\ Provide a SectionProofSlice that proves the given signature for the given parameter forwarded to `knowledge_index`
pub fn prove(
```

Is that what we want? (@madadam could you chip in as well).

Reply:
It's just `node_knowledge_override` is a passing down parameter for the `prove` function, so I think we can skip one of the comment.